src/application/application.cpp

```diff
@@ -25,6 +25,10 @@
 
 #include "predictor.hpp"
 
+#ifdef USE_CUDA
+#include <LightGBM/cuda/vector_cudahost.h>
+#endif
```

Comment:
for all `#include <LightGBM/cuda/vector_cudahost.h>`, maybe `#ifdef USE_CUDA` can be moved inside `LightGBM/cuda/vector_cudahost.h` ?